assets/js/components/surveys/SurveyQuestionMultiSelect.js

```diff
@@ -0,0 +1,238 @@
+			...optionalKeys,
+		};
+	} );
+
+	const initialState = keyBy( mappedChoices, 'answer_ordinal' );
+
+	const [ selectedValues, setSelectedValues ] = useState( initialState );
+
+	// eslint-disable-next-line camelcase
+	const handleCheck = ( answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				selected: ! selectedValues[ answer_ordinal ].selected,
+			},
+		};
+
+		setSelectedValues( newState );
+	};
+
+	// eslint-disable-next-line camelcase
+	const handleAnswerChange = ( event, answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				answer_text: event.target.value?.slice(
+					0,
+					MAXIMUM_CHARACTER_LIMIT
+				),
+			},
+		};
+
+		setSelectedValues( newState );
+	};
+
+	const handleSubmit = () => {
+		const answer = Object.values( selectedValues )
+			.filter( ( { selected } ) => selected )
+			// eslint-disable-next-line camelcase
+			.map( ( { answer_ordinal, answer_text } ) => {
+				// eslint-disable-next-line camelcase
+				if ( answer_text ) {
+					return { answer_ordinal, answer_text };
+				}
+				return { answer_ordinal };
+			} );
+		answerQuestion( {
+			answer,
+		} );
+	};
+
+	const hasEmptySelectedTextValue =
+		// eslint-disable-next-line camelcase
+		choices.filter( ( { write_in, answer_ordinal } ) => {
+			// eslint-disable-next-line camelcase
+			if ( write_in ) {
+				// eslint-disable-next-line camelcase
+				const { selected, answer_text } = selectedValues[
+					answer_ordinal
+				];
+				if ( selected && answer_text.length === 0 ) {
+					return true;
+				}
+			}
+			return false;
+		} ).length > 0;
+
+	const totalSelectedValues = Object.values( selectedValues ).filter(
+		( { selected } ) => selected
+	).length;
+	const hasLessThanMinChoices = totalSelectedValues < minChoices;
+	const hasMoreThanMaxChoices =
+		maxChoices && totalSelectedValues > maxChoices;
+
+	const isSubmitButtonDisabled =
+		hasEmptySelectedTextValue ||
+		hasLessThanMinChoices ||
+		hasMoreThanMaxChoices;
+
+	return (
+		<div className="googlesitekit-survey__multi-select">
+			<SurveyHeader title={ question } dismissSurvey={ dismissSurvey } />
+
+			<div className="googlesitekit-survey__body">
+				{
+					// eslint-disable-next-line camelcase
+					choices.map( ( { answer_ordinal, text, write_in } ) => {
+						const answer = selectedValues[ answer_ordinal ];
+						// eslint-disable-next-line camelcase
+						const uniqueID = `googlesitekit-survey__multi-select-${ answer_ordinal }-${ text }`;
+
+						return (
+							<div
+								key={ text }
+								className="googlesitekit-survey__multi-select__choice"
+							>
+								<Checkbox
+									checked={ answer.selected }
+									onChange={ () =>
+										handleCheck( answer_ordinal )
+									}
+									// eslint-disable-next-line camelcase
+									value={ `${ answer_ordinal }` }
+									id={ text }
```

Comment:
What does this ID end up looking like? Is `text` a long string or just a confusing attribute name? 😆

Reply:
As you wrote on the other PR, this can't have spaces  :smile: 

It's a required prop in `Checkbox` although from the HTML spec it doesn't look required as it's only for labels :thinking: 

For now I've just swapped the white space for dashes. Let me know if there's a better solution!

![Screenshot from 2021-08-11 12-41-41](https://user-images.githubusercontent.com/8496063/129015696-7740fcd6-d8d5-4eb1-879e-b159aaf6dd74.png)